### ISSUE TITLE
DropdownButton options cannot be selected when using TalkBack on Android

### ISSUE THREAD

**Issue by justinmc** [flutter/flutter#168164]
DropdownButton seems completely broken when using TalkBack on Android. I'm unable to select any option. Talkback only highlights the whole screen. I tried this with a physical PixelTablet running Android 15 and a physical Pixel 9 running Android 16 and both had the same issue.

Steps to reproduce:

 1. Run any app with a DropdownButton, such as the one below, on an Android device with TalkBack enabled.
 2. Tap the DropdownButton twice to show the options.
 3. Tap to select any option.

Expected results:

Options can be highlighted in TalkBack with a single tap and selected with a double tap.

Actual results:

The entire screen is highlighted in TalkBack and the options can not be selected.

https://github.com/user-attachments/assets/02191a74-1450-4425-9fe8-e89277daef17

<details>

<summary>Code</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material 3 Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  String? _dropdownValue = 'Option 1'; // For DropdownButton

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Padding(
        padding: const EdgeInsets.all(16.0),
        child: ListView(
          children: <Widget>[
            Text('Dropdown:', style: Theme.of(context).textTheme.titleMedium),
            const SizedBox(height: 8),
            DropdownButton<String>(
              value: _dropdownValue,
              icon: const Icon(Icons.arrow_drop_down),
              elevation: 16,
              isExpanded: true,
              style: TextStyle(color: Theme.of(context).colorScheme.primary),
              underline: Container(
                height: 2,
                color: Theme.of(context).colorScheme.primary,
              ),
              onChanged: (String? newValue) {
                setState(() {
                  _dropdownValue = newValue;
                });
              },
              items: <String>['Option 1', 'Option 2', 'Option 3', 'Option 4']
                  .map<DropdownMenuItem<String>>((String value) {
                return DropdownMenuItem<String>(
                  value: value,
                  child: Text(value),
                );
              }).toList(),
            ),
          ],
        ),
      ),
    );
  }
}
```

</details>

**Comment by darshankawar** [comment]
Thanks for the report @justinmc 
I observed that the reported behavior occurs in master version, as, on latest stable, it works as expected:

<img width="283" alt="Image" src="https://github.com/user-attachments/assets/46b92d40-a532-49f2-aaec-f93c00ffb3d6" />

Verified on master : 3.32.0-1.0.pre.375 and stable : 3.29.3 using S10+ device running Android 11.

**Comment by justinmc** [comment]
Bisected to: https://github.com/flutter/flutter/pull/167522

**Comment by QuncCccccc** [comment]
Reverting #167522. The revert PR is https://github.com/flutter/flutter/pull/168235

**Comment by evageeva** [comment]
JFYI this also affected VoiceOver on iOS, and the associated XCTest APIs like XCUIElement.isHittable. Verified fixed after the rollback.
